saos-webapp/src/main/webapp/WEB-INF/static/javascript/modules/suggester.js

```diff
@@ -12,29 +12,69 @@ var Suggester = (function() {
     var space = {},
     	selItem = [],
     	getUrl = [],
+    	getValue = [],
     	
     	maxElementsInList = 10,
-    	cssClass = "suggestions";
+    	cssClass = "suggestions",
     
+    	SELECTED_FIELD_CLASS = "suggestion",
+    	CONTAINER_ID = "suggestions",
+    	AREA_ID = "suggestion-area",
+    	WRAPPER_ID = "suggestion-wrapper",
+    	
+		FIELD_FOR = "data-field-for",
+    		
+    		
+    	OFFSET = 100;
     
     /* Initialize url parameter and runs main methods: create suggestion container
      * and assign load suggestions methods to input field.
      * 
      * @param $field input field connected to suggester
      * @param source configuration parameters(json):
-     * 		- getUrl - function that takes one @param id and returns url location with list of suggestions.
-     * 		  Module requires function instead of simple url, because this way it can handle url like e.g. "/keywords/COMMON/{phrase}/list"
+     * 		- boxyMode
+     * 			- enabled - turn on boxy mode,
+     * 			- placeholder - text that is used as placeholder in textarea  
+     * 
+     * 		- url - returns url location with list of suggestions.
+     * 		   		it can be function or simple string url, function can handle url like e.g. "/keywords/COMMON/{phrase}/list"
+     *		- getValue - (function) returns value of suggestion list elements e.g.
+     *				function(element) {
+     *					return element.phrase;
+	 *				}
+     *
      */
-    
     init = function($field, source) {
         var fieldId = $field.attr("id");
         
-        if (source.getUrl !== undefined) {
-        	getUrl[fieldId] = source.getUrl;
+        if (source.url !== undefined) {
+        	
+        	if (typeof source.url === "function") {
+        		getUrl[fieldId] = source.url;
+        	} else {
+        		getUrl[fieldId] = function() { return source.url; };
+        	}
         }
-      
-        createContainer($field);
-        assignEvents($field);
+        
+        if (source.getValue !== undefined) {
+        	getValue[fieldId] = source.getValue;
+        }
+        
+        if (source.boxyMode !== undefined && source.boxyMode.enabled !== undefined && source.boxyMode.enabled === true) {
```

Comment:
Unfortunately not, because `source` is a json. If it doesnt have property `source.boxyMode.enabled`, code will crush when we want to check its value.